lib/good_job_connection_pool_size.rb

```diff
@@ -0,0 +1,42 @@
+class GoodJobConnectionPoolSize
+  def self.calculate_worker_pool_size(queues:, cron_enabled:, max_threads:)
+    queues = queues.split(';')
+    # LISTEN/NOTIFY requires 1 connection
+    connections = 1
+    threads = queues.map do |queue|
+      if queue != '*'
+        _name, size = queue.split(':')
+        Integer(size)
+      else
+        0
+      end
+    end.sum
+
+    threads += max_threads if queues.include?('*')
+
+    connections += threads
+
+    # Cron requires two connections
+    connections += 2 if cron_enabled
+
+    connections
+  end
+
+  def self.calculate_primary_pool_size(queues:, max_threads:)
+    queues = queues.split(';')
+    threads = queues.map do |queue|
+      if queue != '*'
+        _name, size = queue.split(':')
+        Integer(size)
+      else
+        0
+      end
+    end.sum
```

Comment:
WDYT of pulling this queue parsing into a helper method? `num_explicit_threads_from_queues` maybe?

Reply:
Yep, I'd like to do that and add tests

Reply:
Added a couple of specs around the less intuitive parts in https://github.com/18F/identity-idp/commit/351f7afc58d10bdc71eb71dc86fe4a9d1108a118